### ISSUE TITLE
Migration dependency setup

### ISSUE THREAD

**Issue by cam-stitt** [cam-stitt/arnold#28]

**Comment by hjc1710** [comment]
Do you have an idea for what you want here?

Sort of like:

```py
__depends__ = ('al2iasdkzxc', 'asdl123023')  # these must be run first
__depended_on_by__ = ('foobar', 'bazquux')  # these must be run after this one

# my migration code here
```

Where if `__depends__` is present, then those other migrations must be migrated first, but it's optional, and, if missing, we fall in line with the file naming conventions?

Or are you thinking of doing what Alembic does with `up_revision`, `down_revision`, and forcing users to maintain a single, linear path for migrations? (If so, that's my least favorite part about Alembic, so I'd request that we hide this behind a flag :smile:).

**Comment by cam-stitt** [comment]
Yes, I was thinking along the line of just having a `__depends__` style argument.
